src/components/new-safe/steps/Step3/index.tsx

```diff
@@ -83,92 +67,90 @@ const CreateSafeStep3 = (): ReactElement => {
   const formMethods = useForm<CreateSafeStep3Form>({
     mode: 'all',
     defaultValues: {
-      [CreateSafeStep3Fields.name]: newSafeName,
-      [CreateSafeStep3Fields.owners]: safeOwners,
-      [CreateSafeStep3Fields.threshold]: safeThreshold,
+      [CreateSafeStep3Fields.name]: data.name,
+      [CreateSafeStep3Fields.owners]: data.owners,
+      [CreateSafeStep3Fields.threshold]: data.threshold,
     },
   })
 
-  const { handleSubmit } = formMethods
+  const { handleSubmit, watch } = formMethods
 
-  const onSubmit = (data: CreateSafeStep3Form) => {
-    console.log(data)
+  const allFormData = watch()
+
+  const handleBack = () => {
+    onBack(allFormData)
```

Comment:
I would recomment calling `getValues` inside `handleBack`. `watch` causes rerenders.

Reply:
True! good suggestion also because I don't expect these values to change in the review step.